### PR TITLE
Add auth and blobstore for image upload 

### PR DESCRIPTION
**Pull request by PratibhaShrivastav**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/user-auth-status")
+public class AuthenticationServlet extends HttpServlet {
+
+    Gson gson = new Gson();
```
Comment (yashjaing) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java]: private staic final GSON

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,14 +19,111 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final String Comment = "Comment";
+  private static final String commentMsgProperty = "commentMsg";
+  private static final String userEmailProperty = "userEmail";
+  private static final String timestampProperty = "timestamp";
+  private static final String imageURLProperty = "imageURL";
+  Gson gson = new Gson();
```
Comment (yashjaing) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: as a java convention all private static finals should be in ALL_CAPS with snake case

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.google.sps.data;
+
+public final class Comment{
+    private final String commentMsg;
+    private final String userEmail;
+    private final long timestamp;
+    private final String imageURL;
```
Comment (yashjaing) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:9]: this looks like a PoJo (simple data holding class with no other methods). You might want to explore lombok (or google protobuf) and builders that they provide.
Much easier to work with and will save you a lot of boilerplate code

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (PratibhaShrivastav) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated code with lombok

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 3 above)
Comment (PratibhaShrivastav) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:9]: Updated code with lombok

Comment (yashjaing) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:9]: use a @NoArgsConstructor with @Getter/@Setter as required and builder to initialise the object same at other places

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -19,14 +19,111 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query(COMMENT).addSort(TIMESTAMP_PROPERTY, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Comment> comments_list = new ArrayList<>();
+    results.asIterable().forEach(entity -> comments_list.add(new Comment((String)entity.getProperty(COMMENT_MSG_PROPERTY),
+      (String)entity.getProperty(USER_EMAIL_PROPERTY),(long)entity.getProperty(TIMESTAMP_PROPERTY), (String)
```
Comment (yashjaing) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: try using the builder as mentioned in prev comments here

**Review by yashjaing — APPROVED** [review]
Some quick changes. Fix and check-in 👍 

Great Work!!

**Comment by PratibhaShrivastav** [comment]
Thanks

**Pull request merged**